services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/ConverterUtils.java

```diff
@@ -128,55 +128,52 @@ private static int findTrimExclusiveEnd(String string, char characterToTrim, int
         }
     }
 
-    public static String[] chunk(String valueToChunk, int... splitSizes) {
-        String[] result = new String[splitSizes.length + 1];
-        int splitStartInclusive = chunkLeft(valueToChunk, result, splitSizes);
+    public static String[] chunkDateTime(String valueToChunk) {
+        Validate.isTrue(valueToChunk.contains("T") && !valueToChunk.contains("+") && !valueToChunk.contains("Z"),
+                        "Invalid sign in this format, expected [-]YYYY-MM-DDTHH:MM:SS.");
```

Comment:
This requires three iterations through the string, can we do it in a single pass?